### PR TITLE
Port nonpublic code

### PR DESCRIPTION
**Pull request by annegambrel**
Ported all the more recent updates to SPIDER private repo into public. May no one ever commit to the SPIDER private repo again.

### CODE REVIEW

File: xfaster/xfaster_class.py [hunk 1]
```diff
@@ -3256,22 +3269,54 @@ def process_index(idx):
...
+
+            if qb_file is None:
+                m_tot = (
+                    scalar + np.sqrt(np.abs(fake_data_r)) * tensor + noise + scaled_temp
+                )
+                self.apply_mask(m_tot, mask)
+                m_alms = self.map2alm(m_tot, self.pol)
+
+            else:  # have to do noise alms separately
+                m_tot = scalar + np.sqrt(np.abs(fake_data_r)) * tensor + scaled_template
```
Comment (arahlin) [inline xfaster/xfaster_class.py:3288]: `scaled_template` -> `scaled_temp` ?

File: xfaster/xfaster_class.py [hunk 2]
```diff
@@ -6899,7 +6962,7 @@ def log_prob(theta):
             self.log(
                 "MCMC converged in {} iterations".format(sampler.iteration), "info"
             )
-            tau = sampler.get_autocorr_time()
+            tau = sampler.get_autocorr_time(tol=0)
```
Comment (arahlin) [inline xfaster/xfaster_class.py:6965]: What does setting `tol=0` do here?

**Review by arahlin — APPROVED** [review]
Looks good!  Should we add some functionality to the example stuff to test it?

File: xfaster/xfaster_class.py (hunk 2 above)
Comment (annegambrel) [reply xfaster/xfaster_class.py:6965]: Unfortunately I didn't write down in detail what that commit was for-- the message was just "avoid error that doesn't save npz file to disk if converged". Docstring for that function says "The minimum number of autocorrelation times needed to trust the estimate." So I guess there was some edge case where things converged given our criteria but didn't meet this criterion and thus it exited with error. I think this is fine though-- if it meets the criteria we spell out, I don't think we need this last one.

Comment (arahlin) [reply xfaster/xfaster_class.py:6965]: Makes sense!

**Comment by annegambrel** [comment]
> Looks good! Should we add some functionality to the example stuff to test it?
I have tested that the example code still works but most of this stuff is not exercised there. Unless screwed up some copy and pasting, it's all been tested in the private repo, and can definitely include use of these functions/options in the more detailed tutorial. I don't think others will be using them until that tutorial is written anyhow since they're not the base case.

**Pull request closed without merging**